packages/get-platform/src/getPlatform.ts

```diff
@@ -375,21 +378,39 @@ export async function getSSLVersion(args: GetOpenSSLVersionParams): Promise<GetO
   return undefined
 }
 
+/**
+ * Get the binary target for the current platform, e.g. `linux-musl-arm64-openssl-3.0.x` for Linux Alpine on arm64.
+ */
 export async function getPlatform(): Promise<Platform> {
-  const { platform } = await getPlatformMemoized()
-  return platform
+  const { binaryTarget } = await getPlatformMemoized()
+  return binaryTarget
+}
+
+export type PlatformWithOSResult = GetOSResult & { binaryTarget: Platform }
+
+function isPlatformWithOSResultDefined(args: Partial<PlatformWithOSResult>): args is PlatformWithOSResult {
+  return args.binaryTarget !== undefined
+}
+
+/**
+ * Get the binary target and other system information (e.g., the libssl version to look for) for the current platform.
+ */
+export async function getPlatformWithOSResult(): Promise<PlatformWithOSResult> {
+  const { memoized, ...rest } = await getPlatformMemoized()
```

Comment:
Doesn't eslint complain about an unused variable here? I vaguely remember that once when I wanted to do something like this I basically did something analogous to `const { memoized: _, ...rest } = await getPlatformMemoized()` and added `_` to the exceptions of the `no-unused-vars` rule in eslint config, but then I had to revert it because someone else found it unintuitive during the code review and asked to use `delete` instead. I think it's a neat pattern though 😄

Reply:
There's no reason to use `delete`, that would really be counterproductive in terms of ergonomics.
Let's go for `const { memoized: _, ...rest }`